app/models/project.rb

```diff
@@ -137,22 +137,22 @@ def to_hash(recurse = true)
       path: UrlGenerator.new.project_path(self),
       hidden: hidden,
       featured: featured,
-      likeCount: likes.count,
+      likeCount: likes.length,
       content: content,
       timeAgoInWords: time_ago_in_words(created_at),
       createdAt: created_at.strftime('%B %d, %Y'),
       ownerName: owner.nil? ? nil : owner.name,
       ownerUrl: owner.nil? ? nil : UrlGenerator.new.user_url(owner),
-      dataSetCount: data_sets.count,
-      dataSetIDs: data_sets.pluck(:id),
-      fieldCount: fields.count,
+      dataSetCount: data_sets.length,
+      dataSetIDs: data_sets.to_a.map(&:id),
```

Comment:
This approach will prevent 2 queries. But on a large project, the single query and object loading will probably be a lot slower than the 2 fast queries that don't do any object loading. However, if this is a recursive serialization then this new approach will be faster.   I suspect you made these changes because when the project is loaded via the query in the `visualizations_controller.rb` with all the includes. In that case this approach doesn't result in any more queries but I think it will still result in lots of object loading.  So a lot of memory will be used when there are a lot of media_objects and data_sets.  I'll submit a PR that should be more efficient but I don't have time to test it.